adr/ADR-22.md

```diff
@@ -0,0 +1,197 @@
+has until now been written to `~/.config/nats/` always.  On Unix systems this
+remains valid; on other systems supporting symbolic links, we should point to
+that location.
+
+The `nats` CLI has proven the utility of the configuration model we're moving
+to.
+
+The configuration used by the `nats` CLI are the configuration contexts, which
+are generic for client tooling and not specific to that tool, so we consider
+the `context/` directory and `context.txt` file to be top-level configuration
+items.
+
+### The `nsc` CLI
+
+The default locations of files for `nsc` can be overridden with the
+environment variables: `$NSC_HOME`, `$NKEYS_PATH`, and `$NSC_CWD_ONLY`.
+
+In addition, `nsc` has traditionally detected an account system Operator in
+the current directory and then ignored all environment variables to use the
+current directory as a root.
+This behavior is considered an authentication administrator mode and should
+not be used by most tools.
+
+Existing client tools should continue to honor `$NSC_HOME` and `$NKEYS_PATH`
+to avoid breaking scripts, but new client tools should not use those
+variables.
+
+These files all now are under the ownership of the `nsc` concept, so live in
+the `nsc/` sub-directory of the NATS locations.
+
+The old file `~/.nsc/nsc.json` is configuration and so would live under
+`$XDG_CONFIG_HOME` or equivalent.  Thus on Unix, `~/.config/nats/nsc/nsc.json`
+should be used.
+
+The credentials and nkeys files are considered data, and so will live under
+`~/.local/share/nats/nsc/stores/` and `~/.local/share/nats/nsc/nkeys/`.
+
+
+### Other environment variables not yet pulled in
+
+Some client tools support `$NATS_KEY` and `$NATS_CERT` to refer to client
+X.509 keys and certificates.
+
+## Consequences
+
+At some point in the future, we should attempt to migrate existing files in
+their old locations to the new standard locations.  We should leave behind
+symbolic links in "the other direction" to prevent breakage.
+
+We should never remove the symbolic links from the old locations, it's up to
+the users/administrators to do so when they're happy to do so.
+
+
+## Registry
+
+* Configuration
+  + The `context/` directory and `context.txt` file are used as per [ADR-21][]
+    for configuration contexts.
+    - Library implementations of contexts should be aware of this.
+    - Most applications should avoid assumptions.
+    - Contexts can contain sensitive data, where a server uses password-based
+      authentication.
+    - compatibility location: strict XDG adherence on all platforms.
+  + The `nsc/` and `nkeys/` directories are reserved for client credentials
+    and the account system, as described above.
+    - Most tools should not need to be specifically aware in code of the
+      location of these; where credentials are passed in, they are typically
+      part of the site configuration.
+    - New tools should use [ADR-21][] configuration contexts to locate
```

Comment:
fyi this repo will change `ADR-21` to a link, so you can take away the markdown reference etc